### PR TITLE
Add Spark Avro JMH tests

### PR DESCRIPTION
**Pull request by shardulm94**
Results from the runs:

```
Benchmark                                                            Mode  Cnt   Score    Error  Units
IcebergSourceFlatAvroDataReadBenchmark.readFileSource                  ss    5  14.364 ± 12.620   s/op
IcebergSourceFlatAvroDataReadBenchmark.readIceberg                     ss    5   9.286 ±  0.297   s/op
IcebergSourceFlatAvroDataReadBenchmark.readWithProjectionFileSource    ss    5   6.072 ±  0.151   s/op
IcebergSourceFlatAvroDataReadBenchmark.readWithProjectionIceberg       ss    5   6.599 ±  0.464   s/op
```

```
Benchmark                                                              Mode  Cnt  Score   Error  Units
IcebergSourceNestedAvroDataReadBenchmark.readFileSource                  ss    5  8.610 ± 0.509   s/op
IcebergSourceNestedAvroDataReadBenchmark.readIceberg                     ss    5  5.518 ± 2.093   s/op
IcebergSourceNestedAvroDataReadBenchmark.readWithProjectionFileSource    ss    5  7.841 ± 0.573   s/op
IcebergSourceNestedAvroDataReadBenchmark.readWithProjectionIceberg       ss    5  5.533 ± 5.309   s/op
```

### CODE REVIEW

**Comment by rdblue** [comment]
Looks good to me! Thanks for adding these.

@aokolnychyi & @danielcweeks, what do you think?

File: build.gradle [hunk 1]
```diff
@@ -251,6 +251,14 @@ project(':iceberg-spark') {
     }
     testCompile project(path: ':iceberg-hive', configuration: 'testArtifacts')
     testCompile project(path: ':iceberg-api', configuration: 'testArtifacts')
+
+    // spark-avro is required only for JMH tests, so ideally we would add this
+    // dependency only to the jmh configuration, however gradle-consistent-versions
+    // plugin does not respect this configuration and does not seem to have a way
+    // to add custom configurations in its lockable configuration detection
+    compileOnly("org.apache.spark:spark-avro_2.11") {
```
Comment (aokolnychyi) [inline build.gradle]: Shall we double-check this with @mccheah?

Comment (mccheah) [reply build.gradle]: This is... strange to me, to say the least. Why not just mark it as `compile`/

Comment (mccheah) [reply build.gradle]: Actually since we do `compileOnly` for `spark-hive` above this is ok from precedent.

**Comment by rdblue** [comment]
I'll merge this since @aokolnychyi had a look and there's just a minor style issue with it.

**Pull request merged**